### DISCUSSION TITLE
change the sampling period(DT)

### DISCUSSION THREAD

**Discussion by clun1** [OpenFAST/openfast#1987] [General]
May I ask if I want to change the sampling time in fast.farm and change the interval of each sampling from 0.01s to 0.1s? Besides modifying the DT parameter in the fst file, what else needs to be modified so that the following error will not be reported
<img width="710" alt="image" src="https://github.com/OpenFAST/openfast/assets/93849049/4d15d1d9-1db1-4fbb-afe2-ea97b0bc7421">

**Comment by jjonkman** [comment]
Dear @clun1.

You can't set the time step (`DT`) in OpenFAST arbitrarily because the required `DT` is tied to the intrinsic dynamics of the wind turbine model.  If you want to change the sampling time of the OpenFAST output, you can set `DT_Out` in the OpenFAST primary (_*.fst_) input file to be an integer multiple of `DT`.  In your case, it sounds like you should revert `DT` back to 0.01 s and set `DT_Out` equal to 0.1 s.

Best regards,

**Comment by clun1** [comment]
Dear jjonkman

Thank you very much for your guidance. My purpose is to change the sampling period in fast.farm, or the interaction time between DLL and fast.farm data from the original 0.01s to 0.1s, and put the wind turbine parameters read in the sampling period 0.1s into MPC (Model Predictive Control) for calculation. But I don't know how to modify the sampling period in fast.farm. Therefore, the DT parameter in the fst file was modified, and the parameter was changed from the original 0.01s to 0.1s, but the error in the above picture would appear. I would like to know how I can modify the DLL to read the data of the fast.farm stroke unit every 0.1s?

Best regards,

**Reply by jjonkman** [reply]
Dear @clun1

OpenFAST communicates with controller DLL's at time step defined by `DLL_DT` in the ServoDyn input file.  It sounds like you want to set `DLL_DT` = 0.1 s, while reverting `DT` back to 0.01 s.

It also sounds like you are using the super controller functionality of FAST.Farm; is that correct?  The super controller will be called by FAST.Farm at a time step of `DT_Low`, which is the same time step used by FAST.Farm to call OpenFAST (but OpenFAST will take smaller time steps of `DT` via subcycling).  Regardless of the value of `DT` or `DLL_DT`, the data passed from the supercontroller to OpenFAST and to the DLL will be updated at a time step of `DT_Low`.

Best regards,

**Reply by clun1** [reply]
Dear jjonkman

Thank you very much. Your answer has helped me a lot. Now I can change the time interval of communication

Best regards,
